MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -1,12 +1,173 @@
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        showFirstQuestion ()
+    }
+    
+    private var currentQuestionIndex: Int = 0
+    private var correctAnswers: Int = 0
+    private var currentQuestion: QuizQuestion {
+        questions[currentQuestionIndex]
+   }
+    
+    // для состояния "Вопрос задан"
+    private struct QuizStepViewModel {
+        let image: UIImage
+        let question: String
+        let questionNumber: String
+    }
+    // для состояния "Результат квиза"
+    private struct QuizResultsViewModel {
+        let title: String
+        let text: String
+        let buttonText: String
+    }
+    
+    private struct QuizQuestion {
+        let image: String
+        let text: String
+        let correctAnswer: Bool
+    }
+    
+    private func showFirstQuestion () {
+        let nextQuestion = questions[currentQuestionIndex]
+        let viewModel = convert(model: nextQuestion)
+        show(quiz: viewModel)
+    }
+    
+    private func show(quiz step: QuizStepViewModel) {
+            imageView.image = step.image
+            counterLabel.text = step.questionNumber
+            textLabel.text = step.question
+    }
+    
+    private func show(quiz result: QuizResultsViewModel) {
+        let alert = UIAlertController(
+            title: result.title,
+            message: result.text,
+            preferredStyle: .alert)
+        
+        let action = UIAlertAction(title: result.buttonText, style: .default) {_ in
+            self.currentQuestionIndex = 0
+            
+            // заново показываем первый вопрос
+            let firstQuestion = self.questions[self.currentQuestionIndex]
+            let viewModel = self.convert(model: firstQuestion)
+            self.show(quiz: viewModel)
+        }
+        
+        alert.addAction(action)
+        
+        self.present(alert, animated: true, completion: nil)
     }
+    
+    private func convert(model: QuizQuestion) -> QuizStepViewModel {
+        return QuizStepViewModel(
+            image: UIImage(named: model.image) ?? UIImage(), // распаковываем картинку
+            question: model.text, // берём текст вопроса
+            questionNumber: "\(currentQuestionIndex + 1)/\(questions.count)") // высчитываем номер вопроса
+    }
+    
+    private func showAnswerResult(isCorrect: Bool) {
+        if isCorrect {
+            correctAnswers += 1
+        }
+        imageView.layer.masksToBounds = true
+        imageView.layer.borderWidth = 8
+        imageView.layer.borderColor = isCorrect ? UIColor.ypGreen.cgColor : UIColor.ypRed.cgColor
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
+            self.imageView.layer.masksToBounds = true
+            self.imageView.layer.borderWidth = 8
+            self.imageView.layer.borderColor =  UIColor.clear.cgColor
+        }
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
+            self.showNextQuestionOrResults()
+        }
```

Comment:
тут стоит писать как одно замыкание, например так:
```suggestion
        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
            self.hideAnswerBorder() 
            self.showNextQuestionOrResults()
        }
```

где hideAnswerBorder - новая функция с содержимым убирающим рамку